oxide_physics/src/pfgen.rs

```diff
@@ -0,0 +1,93 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use crate::particle::*;
+use oxide_math::commons::vector::*;
+use oxide_math::commons::vector3::Vector3;
+use std::vec::Vec;
+
+extern crate generational_arena;
+
+/*
+ * force generator trait
+ */
+pub trait ParticleForceGenerator {
+    /*
+     * interface to calculate and update the force
+     * to the given particle
+     */
+    fn update_force(&self, particle: &mut Particle, duration: f32);
+}
+
+/*
+ * Registry for (Particle, Force Generator) pairs
+ * Holds all particles and associated force generators
+ */
+struct ParticleForceRegistration {
+    particle: DefaultParticleHandle,
+    fg: DefaultForceGeneratorHandle,
+}
+
+pub type DefaultForceGeneratorHandle = generational_arena::Index;
+
+pub struct ParticleForceRegistry {
+    registrations: Vec<ParticleForceRegistration>,
+}
+
+impl ParticleForceRegistry {
+    pub fn add(&self, particle: DefaultParticleHandle, fg: DefaultForceGeneratorHandle) {
+        todo!()
+    }
```

Comment:
This function has no return value?
also, generally `todo!();` is used.

Reply:
I had not considered the return type. Added return type as Result<Self, &str>.
todo!(); macros are all fixed. Thanks!